### ISSUE TITLE
Onset Picker 

### ISSUE THREAD

**Issue by emilyrohrbough** [cerner/terra-clinical#186]
# Issue Description
The Onset Picker is a component that provides users a way to enter or select an approximate date for use in onset scenarios.

## Issue Type

- [x] New Feature
- [ ] Enhancement
- [ ] Bug
- [ ] Other

@doogthemushroom

**Comment by emilyrohrbough** [comment]
# Tech Design 

# Tech Design for Onset Picker
## Summary:
The Onset Picker is a component that provides users a way to enter or select an approximate date for use in onset scenarios.

It will be comprised of controlled select inputs for changing precision (keywords indicating time periods) and granularity (time durations),
which prompts final date selection through additional selects or a date picker.

## Responsiveness:
Not applicable.

## Bidirectionality
Alignment will be mirrored from left to right and right to left.

## Accessibility
Can use keyboard navigation between inputs.
Input components handle their own accessibility.

## React Props:
| Prop     | Type | Default | Description  |
|----------|------|---------|--------------|
| birthdate | string | undefined | Required. The ISO 8601 string representation of the birth date to calculate an onset date from. |
| granularity | string | 'MONTH' | The granularity of the onset date. YEAR, MONTH, and DATE are accepted. |
| granularitySelectName | string | undefined |  Required. Name of the granularity select. The name should be unique. |
| granularitySelectOnChange | function | undefined | A callback function to execute when a granularity is selected. The first parameter is the event. The second parameter is the changed granularity value. |
| precision | string | 'ABOUT' | The precision of the onset date. ABOUT, BEFORE, AFTER, and UNKNOWN are accepted. |
| precisionSelectName | string | undefined | Required. Name of the precision select. The name should be unique. |
| precisionSelectOnChange | function | undefined | A callback function to execute when a precision is selected. The first parameter is the event. The second parameter is the changed precision value. |
| onsetDate | string | moment().format('YYYY-MM-DD') | The ISO 8601 string representation of the onset date to view/modify. Defaults to current date. |
| onsetDateInputName | string | undefined | Required. Name of the onsetDate input. The name should be unique. |
| onsetDateInputOnChange | function | undefined | A callback function to execute when a onsetDate is changed. The first parameter is the event. The second parameter is the changed onsetDate value. |

## Example:
```
<OnsetPicker
  birthdate="2011-08-16T17:47:49-05:00"
  granularitySelectName="my-onset-gran"
  precisionSelectName="my-onset-prec"
  onsetDateInputName="my-onset-date"
/>
```

## CSS Classes:
| Selector         | Description |
|------------------|-------------|
| .terra-OnsetPicker | Class identifier for the onset picker |

**Comment by jmsv6d** [comment]
A few things small things: 
1) Do you have any ux designs for this component for this component that you can post on here?
2) Since we've switched to css modules there is no need to namespace css classes and we've switched our naming convention to having each word separated by a single hyphen
3) The granularity prop type should be `oneOf(['YEAR', 'MONTH', 'DATE', 'AGE'])`And it looks like you excluded the AGE type in your description. 
4) Can you specify in the description that the birthdate is used to calculate the onset date for the AGE precision. It took me a minute to figure out how that prop was relevant.

**Comment by jmsv6d** [comment]
Update the precision prop type to be oneOf as well

**Comment by tbiethman** [comment]
Regarding your callbacks, do you want the first parameter to be the event? If we expose the event, we're going to leak implementation details of our component, like the ref to the target element, through the event API.

**Comment by doogthemushroom** [comment]
@jmsv6d 
1. I do not have any ux designs
2. I can't update @emilyrohrbough comment with the TD, but the css class can be removed as it is not necessary.
3. I have updated this in the code PR
4. Updated in code PR
5. Update precision prop in code PR.

@tbiethman You're right on this. I'm not sure why I was passing the event as it is not a useful value to pass on. This will be changed on the code PR.
